### ISSUE TITLE
Code issue

### ISSUE THREAD

**Issue by BasilLeave** [youngwk/IDMN#1]
Dear author, I am very interested in your excellent work. When trying to reproduce your code, I found that the code for noise injection may be a bit problematic. As shown in the figure below, additive_threshold should depend on neg_logits instead of pos_logits. I would like to ask if there is any problem with my understanding. I look forward to your reply.
![issu](https://github.com/user-attachments/assets/f2a4d941-8053-488c-a220-0b4601503da6)

**Comment by youngwk** [comment]
Hi, thank you for your interest in our work. 

Actually, the code implementation is correct. 

As stated in the paper, the threshold for additive noise (additive_threshold) depends on the positive logits (pos_logits). Please note that Eq. (2) in the paper is constructed using CLIP scores based on the positive labels. 

The reason why the threshold for additive noise depends on the pos_logits rather than the neg_logits is due to the severe positive-negative imbalance in multi-label datasets, with significant variations in the imbalance ratio across different datasets. If we set the additive threshold based on the percentile of neg_logits and the subtractive threshold based on the percentile of pos_logits, it becomes challenging to control the amount of additive noise and subtractive noise. Therefore, we have unified the thresholds based on pos_logits for ease of noise control.
